### PR TITLE
[PNPG-244] feat: added user apis and implementation of onboardingUserPg method

### PR DESCRIPTION
**Pull request by empassaro**
#### List of Changes

added user apis and implementation of onboardingUserPg method

#### Motivation and Context

This method implements the logic required to persist new onboarding which on azure function will execute the addition of the new Manager and the deletion of inactive managers of the PG

#### How Has This Been Tested?

local env

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/onboarding-ms/src/main/java/it/pagopa/selfcare/onboarding/service/OnboardingService.java [hunk 1]
```diff
@@ -27,6 +27,8 @@ public interface OnboardingService {
 
     Uni<OnboardingResponse> onboardingAggregationCompletion(Onboarding onboarding, List<UserRequest> userRequests, List<AggregateInstitutionRequest> aggregates);
 
+    Uni<OnboardingResponse> onboardingUserPg(Onboarding onboarding, List<UserRequest> userRequests);
```
Comment (andrea-putzu) [inline apps/onboarding-ms/src/main/java/it/pagopa/selfcare/onboarding/service/OnboardingService.java:30]: What do you think about calling it "onboardingUserWithoutContract"?

File: apps/onboarding-ms/src/main/java/it/pagopa/selfcare/onboarding/service/OnboardingServiceDefault.java [hunk 2]
```diff
@@ -1278,4 +1291,160 @@ private Uni<OnboardingUtils.ProxyResource> getEC() {
                 .type(EC)
                 .build());
     }
+
+    public Uni<OnboardingResponse> onboardingUserPg(Onboarding onboarding, List<UserRequest> userRequests) {
+        checkOnboardingPgUserList(userRequests);
+        onboarding.setWorkflowType(WorkflowType.USERS_PG);
```
Comment (andrea-putzu) [inline apps/onboarding-ms/src/main/java/it/pagopa/selfcare/onboarding/service/OnboardingServiceDefault.java:1297]: workflow.USER_WITHOUT_CONTRACT?

File: apps/onboarding-ms/src/main/java/it/pagopa/selfcare/onboarding/service/OnboardingServiceDefault.java [hunk 3]
```diff
@@ -1278,4 +1291,160 @@ private Uni<OnboardingUtils.ProxyResource> getEC() {
...
+
+    private Supplier<ResourceNotFoundException> resourceNotFoundExceptionSupplier(Onboarding onboarding) {
+        return () -> new ResourceNotFoundException(String.format("Onboarding not found for taxCode %s, origin %s, productId %s",
+                onboarding.getInstitution().getTaxCode(), onboarding.getInstitution().getOrigin(), onboarding.getProductId()));
+    }
+
+    private Onboarding copyDataFromPreviousToCurrentOnboarding(Onboarding previousOnboarding, Onboarding currentOnboarding) {
+        currentOnboarding.setReferenceOnboardingId(previousOnboarding.getId());
+        currentOnboarding.setInstitution(previousOnboarding.getInstitution());
+        currentOnboarding.setCreatedAt(LocalDateTime.now());
```
Comment (andrea-putzu) [inline apps/onboarding-ms/src/main/java/it/pagopa/selfcare/onboarding/service/OnboardingServiceDefault.java:1346]: This is a new value, this cannot came from oldest onboarding

File: apps/onboarding-ms/src/main/java/it/pagopa/selfcare/onboarding/service/OnboardingServiceDefault.java [hunk 4]
```diff
@@ -1278,4 +1292,160 @@ private Uni<OnboardingUtils.ProxyResource> getEC() {
...
+        String institutionId = currentOnboarding.getInstitution().getId();
+
+        LOG.infof("Checking if user with id: %s is already manager of the institution with id: %s", newManagerId, institutionId);
+
+        return userInstitutionApi.retrieveUserInstitutions(
+                institutionId,
+                null,
+                List.of(currentOnboarding.getProductId()),
+                List.of(PartyRole.MANAGER.name()),
+                List.of("ACTIVE"),
```
Comment (manuraf) [inline apps/onboarding-ms/src/main/java/it/pagopa/selfcare/onboarding/service/OnboardingServiceDefault.java:1382]: you could use RELATIONSHIP_STATE enum or constant

**Review by manuraf — APPROVED** [review]

**Pull request merged**